scrnaseq.Rmd

```diff
@@ -465,6 +546,67 @@ p = PlotMystyle(p, "UMAP, cells coloured by cluster identity", legend_position="
 p
 ```
 
+## Cell Cycle Effect
+How much do gene expression profiles in your dataset reflect the cell cycle phases the single cells were in? We determine the effects of cell cycle heterogeneity by calculating a score for each cell based on its expression of G2M and S phase markers. Scoring is based on the strategy described in `r knitcitations::citet("10.1126/science.aad0501")`, and human gene names are translated using biomaRt. 
+```{r cellCycleEffect}
+# If not done in a previous chunk, get cell cycle genes translated
+if(!param$cc_remove) {
```

Comment:
Why do we do this twice and on two different assays? Should it be like this: Select a normalisation method, judge cell cycle, provide results to user, and then correct?

Reply:
If the cell cycle effect should be removed, one approach is to first do standard normalisation, and then cell cycle scoring. The cell cycle scores can then be added to `vars.to.regress` in `SCTransform`, see https://github.com/satijalab/seurat/issues/1679. For further analyses, we use the `SCT` assay. 

The chunk `cellCycleEffect` is to check for cell cycle effects that are in the data we work with, so either in general if no effects were removed, or after effects were removed. 

My suggestion is to run once without remove cell cycle => you get all diagnostics cell cycle plots including UMAP. If cell cycle is a problem, you run it again and remove cell cycle => ideally, the diagnostics cell cycle plots show that the effect is gone. 

The difficulty to place the code for cell cycle removal was that it should come after standard normalisation and before SCTransform, so we can work with cleaned data right from the start, rather than redoing it all. Hope this is clear.

Reply:
Hmm, if I remember correctly, then Cell Cycle scoring is done based on the raw counts (or maybe the norm counts). How about we do it before normalisation and then include it optionally?

Reply:
Maybe leave it like that. We may have to change it for multi-sample support though

Reply:
Afaik, cell cycle scoring is done on the normalised data, which is why I had done it in between normalisation and scaling. 

However, since we discussed that `RNA` scaled data isn't really used anywhere, I skipped regression in `ScaleData` as it took a long time, and regress out cell cycle effects for `SCT`  in `SCTransform` only. 

So after you suggested that the location of the code is weird, I put back together standard normalisation, scaling, variable features, followed by optional cell cycle scoring and removal in `SCTransform`. Maybe wait for the next commit to see it visually.

Reply:
Good agree